### DISCUSSION TITLE
How to extract mapping between raw and cf-python's coordinate names

### DISCUSSION THREAD

**Discussion by robertjwilson** [NCAS-CMS/cf-python#644] [General]
cf-python does an excellent job of giving proper names to coordinates (time, lon, lat etc.) after using `cf.read`. However, is it possibly to extract how the new coordinate names map to those in the raw netCDF file? 

Is that something that is purely internal or is it preserved somewhere?

**Comment by davidhassell** [comment]
Hi @robertjwilson,

Thanks for the question. `cf.read` preserves all netCDF names and makes them available and editable via the various `nc_*` methods that exist on many cf-python objects. e.g. https://ncas-cms.github.io/cf-python/class/cf.DimensionCoordinate.html#netcdf, https://ncas-cms.github.io/cf-python/class/cf.DomainAxis.html#netcdf, etc.

In this example `f` wasn't read from a file, but the "example fields" come with netCDF information as if they had been:

```python
>>> import cf
>>> f = cf.example_field(0)
>>> x = f.coordinate('X')
>>> x.nc_get_variable()    # will give the name from the file (if read from one)
'lon'
>>> x.nc_set_variable('newlon')   # set a new name
>>> x.nc_get_variable()
'newlon'
>>> cf.write(f, 'test_file.nc')   # writes the X coordinate with netCDF name given by x.nc_get_variable() 
```
If no netCDF variable exists (either because you created the Field ab initio and didn't set one, or you deleted it with `nc_del_variable`) then `cf.write` will make up as nice a name as it can.

Hope that helps,
David

**Comment by robertjwilson** [comment]
Thanks for the quick response @davidhassell. That has solved my problem succicently.

Much appreciated

Robert
